mypy/build.py

```diff
@@ -833,6 +836,11 @@ def compute_hash(text: str) -> str:
     return hashlib.md5(text.encode('utf-8')).hexdigest()
 
 
+def compute_module_hash(path: str) -> str:
+    with open(path, 'r') as f:
+        return compute_hash(f.read())
```

Comment:
That's pretty inefficient -- we read the source code in text mode (using the default encoding) and then `compute_hash()` encodes it back to bytes (using `utf-8`) before hashing.

I also believe that we have the module source code already as an attribute in `State` in most cases so there's not even a need to read it from the file (it will be GC'ed by parse_file()).

Also the source should probably be read using read_with_python_encoding() which does a few tricks.

Finally we don't need this hash to be compatible with the interface hash (which is why compute_hash() exists currently).